wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java

```diff
@@ -523,10 +523,8 @@ public WorkspaceDto addEnvironment(@ApiParam("The workspace id")
         requiredNotNull(newEnvironment, "New environment");
         requiredNotNull(envName, "New environment name");
         relativizeRecipeLinks(newEnvironment);
-        final WorkspaceImpl workspace = workspaceManager.getWorkspace(id);
-        workspace.getConfig().getEnvironments().put(envName, new EnvironmentImpl(newEnvironment));
-        validator.validateConfig(workspace.getConfig());
-        return linksInjector.injectLinks(asDto(workspaceManager.updateWorkspace(id, workspace)), getServiceContext());
+        workspaceManager.addEnvironment(id, envName, newEnvironment);
+        return linksInjector.injectLinks(asDto(workspaceManager.getWorkspace(id)), getServiceContext());
```

Comment:
please elaborate on why environment validation here and further should not be performed